spdx-utils/src/main/kotlin/SpdxExpression.kt

```diff
@@ -236,6 +249,35 @@ class SpdxCompoundExpression(
             SpdxOperator.AND -> left.offersChoice() || right.offersChoice()
         }
 
+    override fun applyChoice(choice: SpdxExpression, subExpression: SpdxExpression): SpdxExpression {
+        if (!subExpression.validChoices().containsAll(choice.validChoices())) {
+            throw InvalidLicenseChoiceException(
+                "$choice is not a valid choice for the subtree expression. Valid choices are: ${validChoices()}."
+            )
+        }
+
+        return replaceMatcherWithChoice(subExpression, choice)
+    }
+
+    private fun replaceMatcherWithChoice(subExpression: SpdxExpression, choice: SpdxExpression): SpdxExpression {
```

Comment:
Is "matcher" a left-over from a renaming? Maybe rename the function to `replaceSubexpressionWithChoice`.